dhall/src/Dhall/Repl.hs

```diff
@@ -251,9 +261,80 @@ options =
   ]
 
 
-completer :: Monad m => Repline.WordCompleter m
-completer _ =
-  return []
+completer
+  :: (Monad m, MonadIO m, MonadState Env m)
+  => Repline.CompleterStyle m
+completer =
+  Repline.Prefix
+    dhallCompleter
+    [ optionsCompleter ]
+
+
+optionsCompleter :: Monad m => (String, Repline.CompletionFunc m)
+optionsCompleter =
+  ( pure optionsPrefix
+  , Repline.listCompleter $ (optionsPrefix :) . fst <$> options @Repl
+  )
+
+
+dhallCompleter
+  :: (Monad m, MonadIO m, MonadState Env m)
+  => Repline.CompletionFunc m
+dhallCompleter ctx@(reversedPrefix, _)
+
+  -- Complete absolute filepaths
+  | "/" `isPrefixOf` word
+  = Repline.fileCompleter ctx
+
+  -- Complete relative filepaths
+  | "./" `isPrefixOf` word
+  = Repline.fileCompleter ctx
+
+  -- Complete environment variables
+  | "env:" `isPrefixOf` word
+  = do
+
+    envs <- fmap fst <$> liftIO getEnvironment
+
+    Repline.listCompleter (map ("env:" ++) envs) ctx
+
+  -- Complete record fields
+  | '.' `elem` word
```

Comment:
I think that already works! :smile: 
The code might not make it obvious though, so maybe I should comments about it. Here `word` is the previous word (for some definition of word), the one right before the cursor and it is not reversed! I am using the beginning of that word to know which type of completion to use. The beginning stays the same no matter how far into the partial completion we are.

Reply:
Okay, I found out what caused my confusion.  What I said was only partially correct: it happens if one field's name is a prefix of another field's name, like this:

```
⊢ :let r = { food = 1, foo = 2 }

r : { food : Natural, foo : Natural }

⊢ r.foo
```

Once it completes `foo` it stops and doesn't attempt to also suggest `food` as a completion

Reply:
Ah yes. I think that might be a shortcoming of Haskeline's autocomplete. Or do you have any suggestion for me to try? I could remove `foo` from the list of suggestions once `foo` is entirely spelled out.

Reply:
Fixed it! Turns out it wasn't a shortcoming of Haskeline, I was looking for the children of `foo` instead of stopping at the children of `r`! :sweat_smile: